ethportal-api/src/types/content_key/beacon.rs

```diff
@@ -0,0 +1,136 @@
+pub enum BeaconContentKey {
+    LightClientBootstrap(LightClientBootstrapKey),
+    LightClientUpdates(LightClientUpdatesKey),
+}
+
+/// Key used to identify a light client bootstrap.
+#[derive(Clone, Debug, Decode, Encode, Eq, PartialEq)]
+pub struct LightClientBootstrapKey {
+    /// Hash of the block.
+    pub block_hash: [u8; 32],
+}
+
+/// Key used to identify a set of light client updates.
+#[derive(Clone, Debug, Decode, Encode, Eq, PartialEq)]
+pub struct LightClientUpdatesKey {
+    pub start_period: u64,
+    pub count: u64,
+}
+
+impl From<&BeaconContentKey> for Vec<u8> {
+    fn from(val: &BeaconContentKey) -> Self {
+        val.as_ssz_bytes()
+    }
+}
+
+impl From<BeaconContentKey> for Vec<u8> {
+    fn from(val: BeaconContentKey) -> Self {
+        val.as_ssz_bytes()
+    }
+}
+
+impl TryFrom<Vec<u8>> for BeaconContentKey {
+    type Error = ContentKeyError;
+
+    fn try_from(value: Vec<u8>) -> Result<Self, Self::Error> {
+        BeaconContentKey::from_ssz_bytes(&value).map_err(|e| ContentKeyError::DecodeSsz {
+            decode_error: e,
+            input: hex_encode(value),
+        })
+    }
+}
+
+impl fmt::Display for BeaconContentKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s = match self {
+            Self::LightClientBootstrap(key) => format!(
+                "LightClientBootstrap {{ block_hash: {} }}",
+                hex_encode_compact(key.block_hash)
+            ),
+            Self::LightClientUpdates(key) => format!(
+                "LightClientUpdates {{ start_period: {}, count: {} }}",
+                key.start_period, key.count
+            ),
+        };
+
+        write!(f, "{s}")
+    }
+}
+
+impl OverlayContentKey for BeaconContentKey {
+    fn content_id(&self) -> [u8; 32] {
+        let mut sha256 = Sha256::new();
+        sha256.update(self.as_ssz_bytes());
+        sha256.finalize().into()
+    }
+
+    fn to_bytes(&self) -> Vec<u8> {
+        let mut bytes: Vec<u8> = Vec::new();
+
+        match self {
+            BeaconContentKey::LightClientBootstrap(key) => {
+                bytes.push(0x00);
+                bytes.extend_from_slice(&key.block_hash);
+            }
+            BeaconContentKey::LightClientUpdates(key) => {
+                bytes.push(0x01);
+                bytes.extend_from_slice(&key.start_period.to_le_bytes());
+                bytes.extend_from_slice(&key.count.to_le_bytes());
+            }
+        }
+
+        bytes
+    }
+}
+
+impl Serialize for BeaconContentKey {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&self.to_hex())
+    }
+}
+
+impl<'de> Deserialize<'de> for BeaconContentKey {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let data = String::deserialize(deserializer)?.to_lowercase();
+        let first_two = &data[..2];
```

Comment:
might be a bit cleaner to use [starts_with](https://doc.rust-lang.org/std/primitive.str.html#method.starts_with)